meta-lxatac-bsp: barebox: update to 2024.09.0

hnez opened the pull request:
Just a routine update.

TODO before merging:

- [x] Test on hardware

hnez committed:
meta-lxatac-bsp: barebox: update to 2024.09.0

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

a3f commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch:
```diff
@@ -707,27 +698,7 @@ index fc3a2386dbb9..0afbb1a1fa4f 100644
- 		interrupt-parent = <&gpioa>;
- 		interrupts = <6 IRQ_TYPE_EDGE_RISING>;
- 
-+		/* Reduce RGMII EMI emissions by reducing drive strength */
```
Why is this being removed?

hnez replied:
This change has made it into the mainline Linux https://github.com/torvalds/linux/commit/7fd195f01ae52871b04b752447f865e1a6661487 and mainline barebox https://github.com/barebox/barebox/commit/a0991ad989b2b678aecf1ba85ee6daad469aff89 .

a3f commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch:
```diff
@@ -346,43 +346,6 @@ index 000000000000..5a07ad7996e6
-+ * Instead drivers are called with the device tree node in action and
-+ * instructed to apply it.
-+ *
-+ * We can add support for this binding for STM32 only as first step, but
```
Why is this being removed? v2024.08.0 doesn't support the new binding either. I [added support recently](https://github.com/barebox/barebox/commit/269dad50c259d2685249d533bc17946cca5001bf), but it is slated for inclusion in v2024.10.0.

hnez replied:
It looks like the respective `fixup!` commits did not make it to the `v2024.03.0/customers/lxa/tac` branch or got force pushed away (sounds like something I would do by accident tbh).
I've made sure to push them to `v2024.03.0/customers/lxa/tac` again and have also included them in the last update of this PR.

a3f commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch:
```diff
@@ -604,17 +604,8 @@ index 8a34d15e9005..1fac9bb60cfb 100644
- 		reg = <0>;
- 
--		reset-gpios = <&gpioa 4 GPIO_ACTIVE_HIGH>;
-+		reset-gpios = <&gpioa 4 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
```
Why is this being removed?

hnez replied:
This change has made it into the mainline Linux https://github.com/torvalds/linux/commit/1d4e8b91d242c4afa1970ca6f7e850aab66bf452 and mainline barebox https://github.com/barebox/barebox/commit/a0991ad989b2b678aecf1ba85ee6daad469aff89 .

a3f commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch:
```diff
@@ -445,17 +408,14 @@ index 524d282a1db8..220d5b2c15e5 100644
  $(call build_stm32mp_image, CONFIG_MACH_LXA_MC1, start_stm32mp157c_lxa_mc1, stm32mp157c-lxa-mc1)
++$(call build_stm32mp_image, CONFIG_MACH_LXA_TAC, start_stm32mp15xc_lxa_tac, stm32mp15xc-lxa-tac)
  
-+pblb-$(CONFIG_MACH_LXA_TAC) += start_stm32mp15xc_lxa_tac
```
This reverts https://github.com/linux-automation/meta-lxatac/commit/e285dad4f06688c9df3f5b1d9720a7db5191542b which is likely not intended.

hnez replied:
It looks like the respective `fixup!` commit did not make it to the `v2024.03.0/customers/lxa/tac` branch or got force pushed away (sounds like something I would do by accident tbh).
I've made sure to push it to `v2024.03.0/customers/lxa/tac` again and have also included it in the last update of this PR.

a3f commented on meta-lxatac-bsp/recipes-core/images/emmc-boot-image/lxatac/genimage.config:
```diff
@@ -18,7 +18,7 @@ image @IMAGE@fip {
         fw-config = "stm32mp157c-lxa-tac-fw-config.dtb"
         hw-config = "stm32mp157c-lxa-tac-fw-config.dtb"
-        nt-fw = "barebox-stm32mp15xc-lxa-tac.img"
+        nt-fw = "barebox-stm32mp15xc-lxa-tac.stm32"
```
The .stm32 image contains at 256 byte header that I suspect will be executed as code. The .img omitted that header.

hnez replied:
My bad. This was a result of me missing the fixup that did the `stm32` --> `img` transition.
The fixup from https://github.com/linux-automation/meta-lxatac/commit/e285dad4f06688c9df3f5b1d9720a7db5191542b is now back in and this change no longer required.

hnez commented:
It looks like I screwed up somewhere while rebasing the changes on top of 2024.08.0. I will have to investigate that a bit.

Since we do not need this update in the next stable release, I will change this to a draft PR and investigate once the release is out.

hnez commented:
I've revisited this PR now and did indeed find a mistake while rebasing - I was missing two fixup commits that changed the image file type and added pull-ups to the button inputs.
This should be fixed now.

While at it I also updated to 2024.09 instead of 2024.08.

hnez commented:
Hi @a3f, could you have a second look at this PR?
And maybe https://github.com/linux-automation/meta-lxatac/pull/183 as well.
